app/models/reports/repository.rb

```diff
@@ -238,18 +239,23 @@ def denominator(region, period)
     #   }
     #
     def region_period_cached_query(calculation, &block)
+      results = regions.each_with_object({}) { |region, hsh| hsh[region.slug] = Hash.new(0) }
       items = cache_entries(calculation)
       cached_results = cache.fetch_multi(*items, force: bust_cache?) { |entry| block.call(entry) }
-      cached_results.each_with_object({}) do |(entry, count), results|
-        results[entry.region.slug] ||= Hash.new(0)
-        next if earliest_patient_recorded_at_period[entry.slug].nil?
-        next if entry.period < earliest_patient_recorded_at_period[entry.slug]
-        results[entry.region.slug][entry.period] = count
-      end
+      cached_results.each { |(entry, count)| results[entry.region.slug][entry.period] = count }
+      results
     end
 
+    # Generate all necessary region period cache entries, only going back to the earliest
+    # patient registration date for Periods. This ensures that we don't create many cache entries
+    # with 0 data for newer Regions.
     def cache_entries(calculation)
-      combinations = regions.to_a.product(periods.to_a)
+      combinations = regions.each_with_object([]) do |region, sum|
```

Comment:
I'm not sure sum is the best name for the array. Maybe just "result" like we used above.

Reply:
👍